onadata/apps/api/tests/viewsets/test_xform_list_viewset.py

```diff
@@ -100,6 +100,65 @@ def test_get_xform_list_form_id_filter(self):
             content = response.render().content.decode('utf-8')
             self.assertEqual(content, form_list_xml % data)
 
+    def test_form_id_filter_for_non_require_auth_account(self):
+        """
+        Test formList formID filter for account that requires authentication
+        """
+        # Bob submit forms
+        xls_path = os.path.join(settings.PROJECT_ROOT, "apps", "main", "tests",
+                                "fixtures", "tutorial.xls")
+        self._publish_xls_form_to_project(xlsform_path=xls_path)
+
+        xls_file_path = os.path.join(settings.PROJECT_ROOT, "apps", "logger",
+                                     "fixtures",
+                                     "external_choice_form_v1.xlsx")
+        self._publish_xls_form_to_project(xlsform_path=xls_file_path)
+
+        # Set require auth to true for this user
+        self.user.profile.require_auth = True
+        self.user.profile.save()
+
+        # Ensure that anonymous users do not have access to private forms
```

Comment:
can we explicitly set `self.xform.shared = False` before this comment so that we are sure that the form being accessed is private?